llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -329,7 +329,12 @@ VPPartialReductionRecipe::computeCost(ElementCount VF,
   // recipe.
   auto HandleWiden = [&](VPWidenRecipe *Widen) {
     if (match(Widen, m_Sub(m_ZeroInt(), m_VPValue(Op)))) {
-      Widen = dyn_cast<VPWidenRecipe>(Op->getDefiningRecipe());
+      Widen = dyn_cast<VPWidenRecipe>(Op);
+      if (!Widen) {
```

Comment:
I left a comment yesterday on @SamTebbs33' patch that took these changes: https://github.com/llvm/llvm-project/pull/147302#discussion_r2444983975

Reply:
Just to clarify what I meant, this if-condition can be removed (although feel free to keep the assert).